third-party-code.rst

```diff
@@ -0,0 +1,253 @@
+########################
+ Third Party Code Usage
+########################
+
+**************
+ Introduction
+**************
+
+The PHP project may leverage code written by others, of which the PHP ecosystem
+has an ample supply. However, it also wants to avoid giving the appearance of
+endorsing or recommending any particular tool over other equally-capable
+competitors, even if unintentionally. This document provides a heuristic and
+process for addressing that balance.
+
+*************
+ Definitions
+*************
+
+**PHP tooling**
+   Refers to the code behind the PHP.net website, the documentation generator
+   project PhD, the PHP wiki, and other similar systems. In general, “PHP code
+   run by PHP.net.”
+
+**Documentation**
+   Refers to objective information about PHP, the PHP language, the PHP standard
+   library, and PHP ecosystem hosted on PHP.net. This may include reference
+   material, tutorials, FAQs, and similar.
+
+**Marketing material**
+   Refers to content on PHP.net or similar sites intended to promote or
+   evangelize PHP the language or ecosystem.
+
+**Libraries**
+   Refers to existing third party code packages or tools, either C extensions or
+   PHP code, maintained by someone other than the PHP Internals team. It also
+   includes command line utilities used primarily by a developer. It may also
+   refer to non-profit PHP ecosystem organizations, such as the PHP Foundation
+   or PHP-FIG.
+
+**Web Application**
+   Refers to a “full” web framework that provides end-to-end web application
+   capabilities, or an installable complete application. It does not refer to
+   command line utilities used primarily by developers building applications.
+
+**Approved license**
+   Refers to a license `approved by the Free Software Foundation as Free
+   Software <https://www.gnu.org/licenses/license-list.en.html>`_, and that is
+   inbound compatible with GPLv3.
+
+*********************
+ Approval Heuristics
+*********************
+
+PHP Tooling
+===========
+
+PHP tooling MAY make use of third party libraries, provided that the library
+meets all of the “Inclusion” criteria, and does not meet any of the “Exclusion”
+criteria.
+
+Inclusion criteria
+------------------
+
+#. The library must have a stable >= 1.0 release, and have had one for at least
+   a year. (This is to ensure it has longevity.)
+#. The library provides targeted, necessary functionality.
+#. The library is a recognized de facto standard, or one of a small number of de
+   facto standards, in its problem space.
+#. The library is available under an Approved License.
+
+Exclusion criteria
```

Comment:
Only thing that I think could be more explicit here, but is sort of implied by the exclusion criteria, is the subsequent removal of a previously included third party code/library/tool if it becomes redundant/unmaintained/etc in the future. Who can raise that issue (anyone on the ML?). What is the process for the removal exactly?

Reply:
Ugh, you ask this just as I open the vote...

Since in practice, "RFC overrides all", an RFC to move a library from explictly-allowed to explicitly-denied would be simple and straightforward.  (That's why I left in spaces for explict-deny.)